redshift_sqlalchemy/dialect.py

```diff
@@ -195,62 +281,222 @@ class RedshiftDialect(PGDialect_psycopg2):
         }),
         (schema.Table, {
             "ignore_search_path": False,
-            'diststyle': None,
-            'distkey': None,
-            'sortkey': None
+            "diststyle": None,
+            "distkey": None,
+            "sortkey": None,
+            "interleaved_sortkey": None,
         }),
     ]
 
+    def __init__(self, *args, **kw):
+        super(RedshiftDialect, self).__init__(*args, **kw)
+        # Cache domains, as these will be static;
+        # Redshift does not support user-created domains.
+        self._domains = None
+
+    @reflection.cache
+    def get_columns(self, connection, table_name, schema=None, **kw):
+        """
+        Return information about columns in `table_name`.
+
+        See :meth:`~sqlalchemy.engine.interfaces.Dialect.get_columns`.
+        """
+        cols = self._get_redshift_columns(connection, table_name, schema, **kw)
+        if not self._domains:
+            self._domains = self._load_domains(connection)
+        domains = self._domains
+        columns = []
+        for col in cols:
+            column_info = self._get_column_info(
+                name=col.name, format_type=col.format_type,
+                default=col.default, notnull=col.notnull, domains=domains,
+                enums=[], schema=col.schema, encode=col.encode)
+            columns.append(column_info)
+        return columns
+
     @reflection.cache
     def get_pk_constraint(self, connection, table_name, schema=None, **kw):
         """
-        Constraints in redshift are informational only. This allows reflection
-        to work.
+        Return information about the primary key constraint on `table_name`.
+
+        See :meth:`~sqlalchemy.engine.interfaces.Dialect.get_pk_constraint`.
+        """
+        constraints = self._get_redshift_constraints(connection, table_name,
+                                                   schema)
+        pk_constraints = [c for c in constraints if c.contype == 'p']
+        if not pk_constraints:
+            return {'constrained_columns': [], 'name': ''}
+        pk_constraint = pk_constraints[0]
+        # The slice here removes the 'PRIMARY KEY(' prefix and the ')' suffix
+        column_text = pk_constraint.condef[13:-1]
+        constrained_columns = column_text.split(', ')
```

Comment:
> Delimited identifiers (also known as quoted identifiers) begin and end with double quotation marks ("). If you use a delimited identifier, you must use the double quotation marks for every reference to that object. The identifier can contain any character other than the double quote itself. Therefore, you can create column or table names with any string, including otherwise illegal characters, such as spaces or the percent symbol.

Looks like `PRIMARY KEY ("foo,bar", colname)` will fail this.